code-challenges.test.js

```diff
@@ -49,6 +102,26 @@ var testArray1 = [3, 7, 10, 5, 4, 3, 3]
 var testArray2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
-
-
+describe("noDuplicate", () => {
+    let testArray1 = [3, 7, 10, 5, 4, 3, 3]
+    let testArray2 = [7, 8, 2, 3, 1, 5, 4]
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it("takes in two arrays as arguments and returns one array with no duplicate values.", () => {
+
+        //an expect method, nested within the test block, calling on the noDuplicate() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+        expect(noDuplicate(testArray1,testArray2)).toEqual([3, 7, 10, 5, 4, 8, 2, 1])
+    })
+})
+// -- Got the red --  
 // b) Create the function that makes the test pass.
+// function noDuplicate(array1, array2) {
+//     // declare the array and use spread operator to conbine the arrays and use [...new Set []] to take out the duplicate values
+//     let newArray =  [...new Set([...array1, ...array2])]
+//     //return newArray 
+//     return newArray
+// }
+// -- Refactor --  
+function noDuplicate(array1, array2) {
+    
+    return newArray =  [...new Set([...array1, ...array2])]
+}
```

Comment:
Nice work. I think the only thing would be removing the newArray. `return [...new Set([...array1, ...array2])]`